source/opt/scalar_replacement_pass.h

```diff
@@ -262,9 +262,31 @@ class ScalarReplacementPass : public MemPass {
   // that we will be willing to split.
   bool IsLargerThanSizeLimit(uint64_t length) const;
 
+  // Copies all relevant decorations from `from` to `to`. This includes
+  // decorations applied to the variable, and to the members of the type.
+  // It is assumed that `to` is a variable that is intended to replace the
+  // `member_index`th member of `from`.
+  void CopyDecorationsToVariable(Instruction* from, Instruction* to,
+                                 uint32_t member_index);
+
+  // Copies pointer related decoration from `from` to `to` if they exist.
+  void CopyPointerDecorationsToVariable(Instruction* from, Instruction* to);
+
+  // Copies the relax precision decoration from the `member_index` of `from` to
+  // `to, if there was one.
+  void CopyRelaxPrecisionDecorationToVariable(Instruction* from,
+                                              Instruction* to,
+                                              uint32_t member_index);
+
+  // Return true if `pointer_type` points to type `id`.  The result will be
+  // false even if `pointer_type` points to a type that is isomorphic to `id`
+  // but not `id`.
+  bool PointsToTypeId(analysis::Pointer* pointer_type, uint32_t id);
```

Comment:
Seems like this is not used? plan to use it in another PR?

Reply:
Good catch. That was part of the old solution, and is no longer needed.